jetty-util/src/main/java/org/eclipse/jetty/util/BufferUtil.java

```diff
@@ -553,10 +553,12 @@ public static void writeTo(ByteBuffer buffer, OutputStream out) throws IOExcepti
         }
         else
         {
-            byte[] bytes = new byte[TEMP_BUFFER_SIZE];
+            byte[] bytes = null;
             while (buffer.hasRemaining())
             {
                 int byteCountToWrite = Math.min(buffer.remaining(), TEMP_BUFFER_SIZE);
+                if (bytes == null)
+                    bytes = new byte[byteCountToWrite];
```

Comment:
Actually slight change would be better.  No need to assign the buffer in the loop:
```suggestion
            byte[] bytes = new byte[Math.min(buffer.remaining(), TEMP_BUFFER_SIZE)];
            while (buffer.hasRemaining())
            {
```